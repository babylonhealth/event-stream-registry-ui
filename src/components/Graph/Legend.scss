@import "../EventStreamRegistry/_variables.scss";

.legend {
  position: absolute;
  bottom: 30px;
  left: $lpad;
  opacity: 0.5;
  pointer-events: none;
  z-index: 100;
}
.legend div{
  display: inline;
  padding-right: 10px;
}
.legend span {
  font-family: "Courier New", sans-serif;
  font-size: 12px;
  padding-right: 8px;
}
.legend i {
  vertical-align: middle;
  position: relative;
  padding-right: 5px;
  top: 2px;
  font-size: 12px !important;
  height: 12px !important;
  width: 12px !important;
}
