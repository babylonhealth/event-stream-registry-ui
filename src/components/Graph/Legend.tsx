import * as React from "react";

import "./Legend.scss";


/**
 * Renders a legend for the Event Stream Registry graph
 * @constructor
 */
const Legend = () => {
    return (
        <div className="legend">
            <div>
                <span>TOPIC :</span>
                <i className="material-icons">fiber_manual_record</i>
            </div>
            <div>
                <span>SERVICE :</span>
                <i className="material-icons">crop_din</i>
            </div>
        </div>
    );
};

export default Legend;
