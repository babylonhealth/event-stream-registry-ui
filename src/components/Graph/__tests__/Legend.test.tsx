import React from 'react'
import {cleanup, render} from '@testing-library/react'
import Legend from '../Legend'

afterEach(cleanup);

it('should take a snapshot', () => {
    const {asFragment} = render(<Legend />);

    // @ts-ignore
    expect(asFragment(<Legend />)).toMatchSnapshot()
});
