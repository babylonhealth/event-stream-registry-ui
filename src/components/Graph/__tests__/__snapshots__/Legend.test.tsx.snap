// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="legend"
  >
    <div>
      <span>
        TOPIC :
      </span>
      <i
        class="material-icons"
      >
        fiber_manual_record
      </i>
    </div>
    <div>
      <span>
        SERVICE :
      </span>
      <i
        class="material-icons"
      >
        crop_din
      </i>
    </div>
  </div>
</DocumentFragment>
`;
